### PR TITLE
[BASE-29] Create internalv2compat library

### PR DESCRIPTION
**Pull request by kylelemons**
## 💸 TL;DR
This PR introduces the hooks necessary for the upcoming v2 compatibility module.
This does not introduce any new user-facing APIs, and it is very clearly indicated that this is not to be used by users.

## 📜 Details
See BASE-29.

## ✅ Checks

- [x] CI tests (if present) are passing
- [x] Adheres to code style for repo

### CODE REVIEW

File: internalv2compat/compat.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+func GlobalLogger() *zap.SugaredLogger {
+	return globalLogger
+}
+
+var v0logDisabled uint64
+
+// SetGlobalLogger allows baseplate v0 to set the global logger.
+func SetGlobalLogger(logger *zap.SugaredLogger) {
+	if atomic.LoadUint64(&v0logDisabled) > 0 {
+		globalLogger.Warnf("Suppressing global log override; use v2 loging instead")
```
Comment (konradreiche) [inline internalv2compat/compat.go:44]:
```suggestion
		globalLogger.Warnf("Suppressing global log override; use v2 logging instead")
```
or possibly a rewording: `ineffectual call to SetGlobalLogger, use OverrideLogger`. V2 logging also has a double connotation at Reddit :slightly_smiling_face:

**Review by konradreiche — CHANGES_REQUESTED** [review]

**Review by konradreiche — APPROVED** [review]

**Review by mterwill — APPROVED** [review]

File: internalv2compat/compat.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+func GlobalLogger() *zap.SugaredLogger {
+	return globalLogger
+}
+
+var v0logDisabled uint64
+
+// SetGlobalLogger allows baseplate v0 to set the global logger.
+func SetGlobalLogger(logger *zap.SugaredLogger) {
+	if atomic.LoadUint64(&v0logDisabled) > 0 {
+		globalLogger.Warn("ineffectual call to SetGlobalLogger; baseplate.go v2 compatibility mode is active")
```
Comment (fishy) [inline internalv2compat/compat.go:44]: we probably want to emit a prometheus counter for this so it can be monitored/alerted?

**Review by fishy — APPROVED** [review]

**Review by pacejackson — APPROVED** [review]

File: internalv2compat/compat.go (hunk 2 above)
Comment (kylelemons) [reply internalv2compat/compat.go:44]: My plan is to emit the prometheus metric from the compat library itself, WDYT?

Comment (fishy) [reply internalv2compat/compat.go:44]: isn't this "the compat library"? do you have a slightly more concrete example of what you have in mind?

Comment (fishy) [reply internalv2compat/compat.go:44]: :fish:

Comment (kylelemons) [reply internalv2compat/compat.go:44]: Discussed offline

**Pull request merged**